freeplane/src/main/java/org/freeplane/view/swing/map/MapViewPane.java

```diff
@@ -1,251 +1,39 @@
+import org.freeplane.view.swing.map.overview.MapOverviewConstants;
+import org.freeplane.view.swing.map.overview.MapOverviewUtils;
+import org.freeplane.view.swing.map.overview.ResizableBorder;
+import org.freeplane.view.swing.map.overview.ResizePanelMouseHandler;
 
-public class MapViewPane extends JPanel implements IFreeplanePropertyListener, IMapChangeListener {
+public class MapViewPane extends JLayeredPane implements IFreeplanePropertyListener, IMapChangeListener {
     private static final long serialVersionUID = 8664710783654626093L;
 
-    private static final Color THUMB_COLOR_BLUE = new Color(0x32_00_00_FF, true);
-    private static final Color THUMB_COLOR_RED = new Color(0x92_FF_00_00, true);
-
-    private class JMapOverviewImage extends JComponent {
-		private static final long serialVersionUID = 1L;
-		private BufferedImage image;
-		
-		JMapOverviewImage(){
-			MapOverviewMouseHandler handler = new MapOverviewMouseHandler();
-            addMouseListener(handler);
-            addMouseMotionListener(handler);
-            addMouseWheelListener(handler);
-		}
-				
-		@Override
-		public Dimension getPreferredSize() {
-			return new Dimension(mapOverviewSize, mapOverviewSize);
-		}
-
-		private BufferedImage snapshot(Graphics2D g) {
-            Rectangle innerBounds = mapView.getInnerBounds();
-            int x = innerBounds.x;
-            int y = innerBounds.y;
-            int width = innerBounds.width;
-            int height = innerBounds.height;
-            int maxSize = Math.max(width, height);
-            final double scaledSize = mapOverviewSize * g.getTransform().getScaleX();
-			double scale = scaledSize / maxSize;
-
-            AffineTransform translation = AffineTransform.getTranslateInstance((maxSize - width) / 2d - x,
-                    (maxSize - height) / 2d - y);
-            AffineTransform transformer = AffineTransform.getScaleInstance(scale, scale);
-            transformer.concatenate(translation);
-
-            int scaledSizeInt = (int)scaledSize;
-			BufferedImage image = new BufferedImage(scaledSizeInt, scaledSizeInt, BufferedImage.TYPE_INT_ARGB);
-            Graphics2D imageG2D = image.createGraphics();
-            try {
-                imageG2D.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BILINEAR);
-                imageG2D.setColor(mapView.getBackground());
-                imageG2D.fillRect(0, 0, scaledSizeInt, scaledSizeInt);
-                imageG2D.transform(transformer);
-                imageG2D.clip(innerBounds);
-                mapView.paintOverview(imageG2D);
-            } finally {
-                imageG2D.dispose();
-            }
-            return image;
-        }		
-		
-
-		@Override
-        protected void paintComponent(Graphics g) {
-			final Graphics2D g2 = (Graphics2D)g;
-			final AffineTransform transform = g2.getTransform();
-			final double scaleX = transform.getScaleX();
```

Comment:
I want to know when or under which environments will the scaleX NOT equal to 1? could you give some examples?

Reply:
On MacOS when 4K monitor is used with resolution 1080 scaleX=scaleY=2

Reply:
:ok_hand: Thanks!